packages/core/src/app/payment/Payment.tsx

```diff
@@ -426,11 +429,12 @@ class Payment extends Component<PaymentProps & WithCheckoutPaymentProps & WithLa
             const order = state.data.getOrder();
             onSubmit(order?.orderId);
         } catch (error) {
-            if (error.type === 'payment_method_invalid') {
+            const handleSubmitError = error as Error & { type: string };
```

Comment:
Avoid as 😄

Reply:
Used a type guard predicate to get over this